### PR TITLE
Github 10633 Update Javadoc comment to mention gradle instead of ant

### LINKED ISSUE #10633: Update java docs in index/TestBackwardsCompatibility class to use gradle and not ant [LUCENE-9593]
In [PR-1733](https://github.com/apache/lucene-solr/pull/1733) we noticed that the comments for generating old indexes in `org.apache.lucene.index.TestBackwardsCompatibility` use ant instead of gradle.

However, since support for ant has been removed we should update the docs to reflect the new command.

New tests in the same [PR-1733](https://github.com/apache/lucene-solr/pull/1733) suggest that the change in the comments is simple:

Replace `ant test` by `gradlew test`. (with no change in the JVM args)

I've verified that this works for the tests in [PR-1733](https://github.com/apache/lucene-solr/pull/1733)

---
Migrated from [LUCENE-9593](https://issues.apache.org/jira/browse/LUCENE-9593) by Gautam Worah (@gautamworah96)

### PR DESCRIPTION
**Pull request by shaikhu**
I believe this is a fix for issue [10633](https://github.com/apache/lucene/issues/10633).

Although the path is slightly different, this is the only class I could find with the name `TestBackwardsCompatibility`.

### CODE REVIEW

File: lucene/backward-codecs/src/test/org/apache/lucene/backward_index/TestBackwardsCompatibility.java [hunk 1]
```diff
@@ -147,9 +147,10 @@ public class TestBackwardsCompatibility extends LuceneTestCase {
   //
   // -----
   //
-  // To generate backcompat indexes with the current default codec, run the following ant command:
-  //  ant test -Dtestcase=TestBackwardsCompatibility -Dtests.bwcdir=/path/to/store/indexes
-  //           -Dtests.codec=default -Dtests.useSecurityManager=false
+  // To generate backcompat indexes with the current default codec, run the following gradle
+  // command:
+  //  gradlew test -Dtestcase=TestBackwardsCompatibility -Dtests.bwcdir=/path/to/store/indexes
```
Comment (zhaih) [inline lucene/backward-codecs/src/test/org/apache/lucene/backward_index/TestBackwardsCompatibility.java:152]: I don't think `-Dtestcase` works with gradle, please refer to this: https://stackoverflow.com/questions/22505533/how-to-run-only-one-unit-test-class-using-gradle

Also maybe give the command a try to verify your change?

**Review by zhaih — CHANGES_REQUESTED** [review]

File: lucene/backward-codecs/src/test/org/apache/lucene/backward_index/TestBackwardsCompatibility.java (hunk 1 above)
Comment (dweiss) [reply lucene/backward-codecs/src/test/org/apache/lucene/backward_index/TestBackwardsCompatibility.java:152]: Even better, read this help guide:
https://github.com/apache/lucene/blob/main/help/tests.txt#L51-L69

Comment (shaikhu) [reply lucene/backward-codecs/src/test/org/apache/lucene/backward_index/TestBackwardsCompatibility.java]: > I don't think `-Dtestcase` works with gradle, please refer to this: https://stackoverflow.com/questions/22505533/how-to-run-only-one-unit-test-class-using-gradle
> 
> Also maybe give the command a try to verify your change?

Yes, you're correct it `-Dtestcase` doesn't work with Gradle. Testing locally the following command worked
```gradlew test --tests  TestBackwardsCompatibility```.

Comment (shaikhu) [reply lucene/backward-codecs/src/test/org/apache/lucene/backward_index/TestBackwardsCompatibility.java]: > Even better, read this help guide: https://github.com/apache/lucene/blob/main/help/tests.txt#L51-L69

Thanks. See 886a009

**Review by zhaih — CHANGES_REQUESTED** [review]

File: lucene/backward-codecs/src/test/org/apache/lucene/backward_index/TestBackwardsCompatibility.java [hunk 2]
```diff
@@ -227,7 +228,7 @@ public void testCreateMoreTermsIndex() throws Exception {
     Thread.sleep(100000);
   }
 
-  // ant test -Dtestcase=TestBackwardsCompatibility -Dtestmethod=testCreateSortedIndex
+  // gradlew test -Dtestcase=TestBackwardsCompatibility -Dtestmethod=testCreateSortedIndex
```
[the remark this answers is not in the document]
Comment (shaikhu) [reply lucene/backward-codecs/src/test/org/apache/lucene/backward_index/TestBackwardsCompatibility.java]: > Here needs to be updated as well?

Missed that! See 7a75d797fe

**Review by zhaih — APPROVED** [review]
LGTM, thanks! Please add an entry to CHANGES.txt :)

**Comment by shaikhu** [comment]
> LGTM, thanks! Please add an entry to CHANGES.txt :)

Added entry with f81956b8

File: lucene/CHANGES.txt [hunk 3]
```diff
@@ -2129,6 +2129,11 @@ Bug Fixes
 ---------------------
 (No changes)
 
+Documentation
+---------------------
+
+* GITHUB#10633: Update javadocs in TestBackwardsCompatibility to use gradle and not ant. (Usman Shaikh)
```
Comment (zhaih) [inline lucene/CHANGES.txt:2135]: Could you move it under Lucene 9.6?

**Review by zhaih — CHANGES_REQUESTED** [review]

File: lucene/CHANGES.txt (hunk 3 above)
Comment (shaikhu) [reply lucene/CHANGES.txt]: > Could you move it under Lucene 9.6?

moved with 4efae16f93

**Review by zhaih — APPROVED** [review]

**Comment by zhaih** [comment]
Thank you! Just merged and backported

**Pull request merged**